kogito-codegen/src/main/java/org/kie/kogito/codegen/decision/DecisionRestResourceGenerator.java

```diff
@@ -150,6 +161,66 @@ public String generate() {
         return clazz.toString();
     }
 
+    private void processOASAnn(MethodDeclaration dmnMethod, DecisionService ds) {
+        String inputRef = null;
+        String outputRef = null;
+        if (withOASResult!= null) {
+            DMNModelIOSets ioSets = withOASResult.lookupIOSetsByModel(dmnModel);
+            DMNType identifyInputSet = ds != null ? ioSets.lookupDSIOSetsByName(ds.getName()).getDSInputSet() : ioSets.getInputSet();
+            DMNType identifyOutputSet = ds != null ? ioSets.lookupDSIOSetsByName(ds.getName()).getDSOutputSet() : ioSets.getOutputSet();
+            inputRef = withOASResult.getNamingPolicy().getRef(identifyInputSet);
+            outputRef = withOASResult.getNamingPolicy().getRef(identifyOutputSet);
+        }
+        // MP / Quarkus
+        processAnnForRef(dmnMethod,
+                         "org.eclipse.microprofile.openapi.annotations.parameters.RequestBody",
+                         "org.eclipse.microprofile.openapi.annotations.media.Schema",
+                         "/dmnDefinitions.json" + inputRef,
+                         !mpAnnPresent);
+        processAnnForRef(dmnMethod,
+                         "org.eclipse.microprofile.openapi.annotations.responses.APIResponse",
+                         "org.eclipse.microprofile.openapi.annotations.media.Schema",
+                         "/dmnDefinitions.json" + outputRef,
+                         !mpAnnPresent);
+        // io.swagger / SB
+        processAnnForRef(dmnMethod,
+                         "io.swagger.v3.oas.annotations.parameters.RequestBody",
+                         "io.swagger.v3.oas.annotations.media.Schema",
+                         "/docs/dmnDefinitions.json" + inputRef,
+                         !swaggerAnnPresent);
+        processAnnForRef(dmnMethod,
+                         "io.swagger.v3.oas.annotations.responses.ApiResponse",
+                         "io.swagger.v3.oas.annotations.media.Schema",
+                         "/docs/dmnDefinitions.json" + outputRef,
+                         !swaggerAnnPresent);
+    }
+
+    private void processAnnForRef(MethodDeclaration dmnMethod, String parentName, String innerName, String ref, boolean removeIt) {
+        List<NormalAnnotationExpr> findAll = dmnMethod.findAll(NormalAnnotationExpr.class, x -> x.getName().toString().equals(parentName));
+        if (findAll.isEmpty()) {
+            if (removeIt) {
+                return; // nothing to do
+            } else {
+                throw new IllegalStateException();
```

Comment:
What about something like
```suggestion
                throw new IllegalStateException("Impossible to find annotation " + parentName + " on method " + dmnMethod.toString());
```